src/shared/conf-parser.c

```diff
@@ -1389,6 +1390,68 @@ int config_parse_warn_compat(
         return 0;
 }
 
+int config_parse_compression(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        CompressionArgs *args = ASSERT_PTR(data);
+        int r;
+
+        if (isempty(rvalue))
+                return 1;
```

Comment:
Clear previous assignment when an empty string is specified.

Reply:
done